tsl/src/remote/connection.c

```diff
@@ -397,7 +398,7 @@ handle_conn_destroy(PGEventConnDestroy *event)
 	unsigned int results_count = 0;
 	ListNode *curr;
 
-	Assert(NULL != conn);
+	Ensure(conn, "no instance data available");
```

Comment:
I think this really should be an `Assert()` because it checks something that shouldn't happen. PQinstanceData just literally gets a pointer from an array and will only fail if it is not found or a passed in pointer is NULL.

The check should be elsewhere, e.g., where the allocation of the instance data occurs. If we follow this pattern, we will just have unused error messages every time we reference a pointer.